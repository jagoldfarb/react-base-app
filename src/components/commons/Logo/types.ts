export interface ILogoProps {
  isDarkModeOn?: boolean;
}
