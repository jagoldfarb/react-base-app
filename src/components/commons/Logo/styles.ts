import styled from 'styled-components';

import { COLORS } from 'helpers/enums/colors';

import { ILogoProps } from './types';

export const LogoContainer = styled.div`
  a {
    color: ${({ isDarkModeOn }: ILogoProps) => (isDarkModeOn ? COLORS.white : COLORS.black)};
    font-size: 2em;
  }

  a:hover {
    color: ${COLORS.gray};
  }
`;
