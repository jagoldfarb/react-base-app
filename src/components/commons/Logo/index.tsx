import React, { FunctionComponent } from 'react';
import { Link } from 'react-router-dom';

import { LogoContainer } from './styles';
import { ILogoProps } from './types';

const Logo: FunctionComponent<ILogoProps> = (props: ILogoProps) => {
  const { isDarkModeOn = false } = props;

  return (
    <LogoContainer data-testid="logo" isDarkModeOn={isDarkModeOn}>
      <Link to="/home">Logo</Link>
    </LogoContainer>
  );
};

export default Logo;
