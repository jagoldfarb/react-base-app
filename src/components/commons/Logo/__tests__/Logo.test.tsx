import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import Logo from '..';

const defaultProps = {
  isDarkModeOn: false,
};

describe('Logo Component', () => {
  it('Should render the logo with dark mode off', () => {
    const { asFragment, getByTestId } = render(
      <MemoryRouter>
        <Logo {...defaultProps} />
      </MemoryRouter>,
    );
    expect(getByTestId('logo')).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });

  it('Should render the logo with dark mode on', () => {
    const props = { ...defaultProps, isDarkModeOn: true };
    const { asFragment, getByTestId } = render(
      <MemoryRouter>
        <Logo {...props} />
      </MemoryRouter>,
    );
    expect(getByTestId('logo')).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });
});
