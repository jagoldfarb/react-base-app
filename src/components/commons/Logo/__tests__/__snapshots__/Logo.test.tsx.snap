// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo Component Should render the logo with dark mode off 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ fajWDP"
    data-testid="logo"
  >
    <a
      href="/home"
    >
      Logo
    </a>
  </div>
</DocumentFragment>
`;

exports[`Logo Component Should render the logo with dark mode on 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ kCPbJH"
    data-testid="logo"
  >
    <a
      href="/home"
    >
      Logo
    </a>
  </div>
</DocumentFragment>
`;
